scripts/quests/crystalWar/The_Lost_Book.lua

```diff
@@ -0,0 +1,203 @@
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and player:getMainLvl() >= 30 and
+                player:hasKeyItem(xi.ki.BRONZE_RIBBON_OF_SERVICE)
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Door_Rhinostery_South'] = quest:progressEvent(143),
+
+            onEventFinish =
+            {
+                [143] = function(player, csid, option, npc)
+                    quest:begin(player)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+    },
+
+    -- Bring a Mythril Beastcoin and travel to present-day Giddeus and trade Quu Bokye
+    -- a Mythril Beastcoin to obtain a Leather-bound Book.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 1
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Door_Rhinostery_South'] = quest:progressEvent(148), -- Optional dialogue
+        },
+
+        [xi.zone.GIDDEUS] =
+        {
+            ['Quu_Bokye'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, xi.items.MYTHRIL_BEASTCOIN) then
+                        return quest:progressEvent(65)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(64)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [65] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    npcUtil.giveKeyItem(player, xi.ki.LEATHERBOUND_BOOK)
+                    quest:messageSpecial(giddeusID.text.KEYITEM_OBTAINED, xi.ki.LEATHERBOUND_BOOK)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+            },
+        },
+    },
+
+    -- Return to the Rhinostery in Windurst (S), and click the southern door for another cutscene.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 2
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Door_Rhinostery_South'] = quest:progressEvent(144),
+
+            onEventFinish =
+            {
+                [144] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+            },
+        },
+    },
+
+    -- Go to the Optistery (F-8 on the first map) of Windurst (S) and examine the door for a cutscene.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 3
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Door_Optistery']        = quest:progressEvent(145),
+            ['Door_Rhinostery_South'] = quest:progressEvent(150), -- Optional dialogue
```

Comment:
If this is a door that doesn't open, this doesn't need to be a progressEvent. If it does: ignore me

Reply:
It's an otherwise dead NPC I believe. Triggers a reminder cutscene. Will swap to event.